src/components/TopicPage.jsx

```diff
@@ -0,0 +1,54 @@
+import React, { useEffect, useState } from "react";
+import { useParams, Link } from "react-router-dom";
+import { getArticles } from "../../utils/api";
+import ArticleCard from "./ArticleCard";
+import Loading from "./Loading";
+
+const TopicPage = () => {
+  const { topic_id } = useParams();
+  const [articles, setArticles] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    setError(null);
+    getArticles()
+      .then((data) => {
+        const filteredArticles = data.articles.filter(
```

Comment:
You have an endpoint that supports sorting by topic. Why aren't you using it? 

In your api there are 12 articles for cooking. Your front end is only serving me 2. I would strongly recommend coming back to this ticket and making use of your api fully. You should not have the client sorting data as that can lead to stale and incorrect serving of data, as evidenced that I'm not getting the correct amount of articles.